x/subkeys/internal/ante/ante.go

```diff
@@ -0,0 +1,29 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authAnte "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	authKeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authTypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/subkeys/internal/keeper"
+)
+
+// NewAnteHandler is just like auth.NewAnteHandler, except we use the DeductDelegatedFeeDecorator
+// in order to allow payment of fees via a delegation.
+func NewAnteHandler(ak authKeeper.AccountKeeper, supplyKeeper authTypes.SupplyKeeper, dk keeper.Keeper, sigGasConsumer authAnte.SignatureVerificationGasConsumer) sdk.AnteHandler {
```

Comment:
All but two of the entries are decorators from the auth library. This is how I create new logic and reuse code.

I think trying to edit in-place (replace item 5 with X) is much more fragile and hard to reason.

Also note that I needed to change some order on pubkey checks, as I do want to allow a non-existing account with no balance to sign a tx if it has a valid delegation. This order allows the DeductDelegatedFeeDecorator to create a default account if there is a valid delegation and the grantee (signer) has no account. Note the tests to this effect in `ante/fee_test.go`. I found out that the stack and the single decorator had different behavior on a few cases and worked to bring them into alignment.

Reply:
See the comment:

```go
		// DeductDelegatedFeeDecorator will create an empty account if we sign with no tokens but valid validation
		// This must be before SetPubKey, ValidateSigCount, SigVerification, which error if account doesn't exist yet
		NewDeductDelegatedFeeDecorator(ak, supplyKeeper, dk),
```

Reply:
I see! However, my question still kind of stands. Instead of providing a new ante-handler chain, why not just have `NewDeductDelegatedFeeDecorator`. Let the application decide how to chain them. We only have the default chain in `x/auth` as a common default & reference.

Reply:
Okay, then I will make this stack private and just use it for testing.
It is important for me to make sure this decorator plays well with the rest of the stack (and I found and corrected an issue doing so)